src/openvic-simulation/politics/NationalFocus.cpp

```diff
@@ -0,0 +1,118 @@
+
+inline bool NationalFocusManager::add_national_focus_group(std::string_view identifier) {
+	if (identifier.empty()) {
+		Logger::error("No identifier for national focus group!");
+		return false;
+	}
+	return national_focus_groups.add_item({ identifier });
+}
+
+inline bool NationalFocusManager::add_national_focus(
+	std::string_view identifier,
+	std::string_view icon,
+	NationalFocusGroup const& group,
+	ModifierValue&& modifiers,
+	NationalFocus::pop_promotion_map_t&& encouraged_promotion,
+	NationalFocus::party_loyalty_map_t&& encouraged_loyalty,
+	NationalFocus::production_map_t&& encouraged_production
+) {
+	if (identifier.empty()) {
+		Logger::error("No identifier for national focus!");
+		return false;
+	}
+	if (icon.empty()) {
+		Logger::error("No icon for national focus ", identifier);
+		return false;
+	}
+	return national_foci.add_item({ identifier, icon, group, std::move(modifiers), std::move(encouraged_promotion), std::move(encouraged_loyalty), std::move(encouraged_production) });
+}
+
+bool NationalFocusManager::load_national_foci_file(PopManager const& pop_manager, IdeologyManager const& ideology_manager, GoodManager const& good_manager, ModifierManager const& modifier_manager, ast::NodeCPtr root) {
+	bool ret = expect_dictionary_reserve_length(national_focus_groups, [this](std::string_view identifier, ast::NodeCPtr node) -> bool {
+		return add_national_focus_group(identifier);
+	})(root);
+	lock_national_focus_groups();
+
+	ret &= expect_national_focus_group_dictionary([this, &pop_manager, &ideology_manager, &good_manager, &modifier_manager](NationalFocusGroup const& group, ast::NodeCPtr node) -> bool {
+		bool ret = expect_dictionary([this, &group, &pop_manager, &ideology_manager, &good_manager, &modifier_manager](std::string_view identifier, ast::NodeCPtr node) -> bool {
+			std::string_view icon;
+			ModifierValue modifiers;
+			NationalFocus::pop_promotion_map_t promotions;
+			NationalFocus::party_loyalty_map_t loyalties;
+			NationalFocus::production_map_t production;
+
+			Ideology const* last_specified_ideology = nullptr; // weird, I know
+
+			bool ret = modifier_manager.expect_modifier_value_and_keys_and_default(
+				move_variable_callback(modifiers),
+				[&promotions, &pop_manager, &production, &good_manager, &modifiers, &modifier_manager](std::string_view key, ast::NodeCPtr value) -> bool {
+					PopType const* pop_type = pop_manager.get_pop_type_by_identifier(key);
+					if (pop_type != nullptr) {
+						fixed_point_t boost;
+						bool ret = expect_fixed_point(assign_variable_callback(boost))(value);
+						promotions[pop_type] = boost;
+						return ret;
+					}
+					Good const* good = good_manager.get_good_by_identifier(key);
+					if (good != nullptr) {
+						fixed_point_t boost;
+						bool ret = expect_fixed_point(assign_variable_callback(boost))(value);
+						production[good] = boost;
+						return ret;
+					}
+					return false;
```

Comment:
```suggestion
					return key_value_invalid_callback(key, value);
```
OK this revealed the error, invalid key `railroads` (modifier affecting capitalist chance to build railroads). Add `ret &= add_modifier_effect("railroads", true);` to `Modifier.cpp` and everything loads successfully.
In general everywhere you `return false;` should have an error message so we know where the failure occurred, here `key_value_invalid_callback` is basically a prepackaged error message.